### PR TITLE
Intellij Plugin: Implement registered name codeVision hint

### LINKED ISSUE #195: Add debug hint for name conversion
Follow up to #181 
Add name conversion hint to functions/properties/signals to indicate what the converted name is in godot.
Best case would be something similar to what rider is doing with function usages:
115973179-65940480-a553-11eb-9b84-7a2b11e64ab8
Otherwise a new gutter icon should do the trick.

### PR DESCRIPTION
**Pull request by chippmann**
This resolves #195 by implementing so called codeVision hints for registered elements:

And on click on the corresponding codeVision, the converted name is copied to the clipboard:

### CODE REVIEW

**Review by piiertho — APPROVED** [review]

File: kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/GodotCodeVision.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtProperty
+import java.awt.datatransfer.StringSelection
+import javax.swing.JPanel
+
+@Suppress("UnstableApiUsage")
+class GodotCodeVision : InlayHintsProvider<NoSettings> {
```
Comment (raniejade) [inline kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/GodotCodeVision.kt:34]: nit: The name is a generic - maybe `RegisteredNameInlayHint`?

**Review by raniejade — APPROVED** [review]

File: kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/RegisteredNameInlayHint.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import org.jetbrains.kotlin.psi.KtProperty
+import java.awt.Point
+import java.awt.datatransfer.StringSelection
+import java.awt.event.MouseEvent
+import javax.swing.JPanel
+
+@Suppress("UnstableApiUsage")
+class RegisteredNameInlayHint : InlayHintsProvider<NoSettings> {
+
+    override val key: SettingsKey<NoSettings> = SettingsKey("GodotCodeVision")
```
Comment (raniejade) [inline kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/RegisteredNameInlayHint.kt:39]: missed from rename? :)

Comment (chippmann) [reply kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/RegisteredNameInlayHint.kt:39]: indeed
Will fix asap

Comment (chippmann) [reply kt/plugins/godot-intellij-plugin/src/main/kotlin/godot/intellij/plugin/codevision/RegisteredNameInlayHint.kt]: done

**Pull request merged**